Simplified/NYPLBarcodeScanningViewController.m

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright 2012 ZXing authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+@import PureLayout;
+#import <AudioToolbox/AudioToolbox.h>
+#import "NYPLBarcodeScanningViewController.h"
+
+@interface NYPLBarcodeScanningViewController ()
+
+@property (nonatomic, strong) ZXCapture *capture;
+@property (nonatomic, copy) void (^completion)(NSString *resultString);
+
+@end
+
+@implementation NYPLBarcodeScanningViewController {
+	CGAffineTransform _captureSizeTransform;
+}
+
+- (instancetype)initWithCompletion:(void (^)(NSString *resultString))completion
+{
+  self = [super init];
+  if (self) {
+    self.completion = completion;
+    return self;
+  } else {
+    return nil;
+  }
+}
+
+#pragma mark - View Controller Methods
+
+- (void)dealloc {
+  [self.capture.layer removeFromSuperlayer];
+}
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+
+  UIBarButtonItem *cancelButton = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemCancel
+                                                                                target:self
+                                                                                action:@selector(didSelectCancel)];
+  self.navigationItem.leftBarButtonItem = cancelButton;
+
+  self.capture = [[ZXCapture alloc] init];
+  self.capture.camera = self.capture.back;
+  self.capture.focusMode = AVCaptureFocusModeContinuousAutoFocus;
+  [self.view.layer addSublayer:self.capture.layer];
+
+  UIView *previewRect = [[UIView alloc] init];
+  previewRect.layer.borderColor = [UIColor redColor].CGColor;
+  previewRect.layer.borderWidth = 4.0;
+  previewRect.layer.cornerRadius = 20.0;
+  [self.view addSubview:previewRect];
+  [previewRect autoCenterInSuperview];
+  if (self.traitCollection.horizontalSizeClass == UIUserInterfaceSizeClassRegular) {
+    [previewRect autoSetDimensionsToSize:CGSizeMake(350, 170)];
+  } else {
+    [previewRect autoSetDimensionsToSize:CGSizeMake(250, 120)];
+  }
+}
+
+- (void)viewWillAppear:(BOOL)animated {
+  [super viewWillAppear:animated];
+
+  self.capture.delegate = self;
+  [self applyOrientation];
```

Comment:
SimplyE orientation is locked to portrait, am I correct?

Reply:
Correct. This code allows the modal view handling the video scanning to interpret a barcode at any orientation.